src/ovms.h

```diff
@@ -124,6 +125,36 @@ void OVMS_StatusDelete(OVMS_Status* status);
 OVMS_Status* OVMS_StatusGetCode(OVMS_Status* status,
     uint32_t* code);
 
+// Get value of metadata field by json pointer
```

Comment:
Feedback for API call names:

```
OVMS_GetServerMetadata
OVMS_ServerMetadataDelete
```
Those are main methods to create and destroy "ServerMetadata". Naming follows convention for other C-API calls (for example OVMS_GetServableMetadata/OVMS_ServableMetadataDelete)

```
OVMS_GetMetadataByPointer
OVMS_SerializeMetadataToJson
```
Those are utility methods for ServerMetadata object access. You can see it by looking at first parameter (it must be OVMS_ServerMetadata), but the name does not reflect it.
If we decide to add another metadata in the future, and we will also use JSON format, we will not be able to re-use this call since first parameter type differs and the name will be already taken.

My suggestion is to either:
a) rename those calls to reflect container name: `OVMS_GetServerMetadatabyPointer`/`OVMS_SerializeServerMetadataToJson` - sounds a bit too descriptive
b) go for triton serving approach - generic type `Metadata` instead of `ServerMetadata` which would be reused for all metadatas underneath represented as JSON (atm only ServerMetadata). We could reuse utilities this way and reduce amount of C-API calls.

```
OVMS_Free
```
I'm afraid that users could mistake this call for other memory release functions, like:
```
auto* server = OVMS_ServerNew();
OVMS_Free(server);
```
Maybe we could name it `OVMS_StringFree`?